backend/deident/src/main/kotlin/com/arconsis/deident/services/faker/FakerMemoryApi.kt

```diff
@@ -0,0 +1,34 @@
+package com.arconsis.deident.services.faker
+
+import com.benasher44.uuid.Uuid
+import jakarta.enterprise.context.ApplicationScoped
+
+/*
+ * "memory" maps session ids to two maps:
+ *      .first --> maps original to fake value
+ *      .second --> maps fake value to original
+ */
+@ApplicationScoped
+class FakerMemoryApi {
+    private val memory = mutableMapOf<Uuid, Pair<MutableMap<String, String>, MutableMap<String, String>>>()
+
+    fun setFakeValue(sessionId: Uuid, originalValue: String, fakeValue: String) {
+        val session = memory.getOrPut(sessionId) { Pair(mutableMapOf(), mutableMapOf()) }
+        session.first[originalValue] = fakeValue
+        session.second[fakeValue] = originalValue
+        memory[sessionId] = session
+    }
+
+    fun getFakeFromOriginal(sessionId: Uuid, originalValue: String): String {
+        val session = memory[sessionId] ?: throw Exception("Could not read faker memory for session id: $sessionId")
```

Comment:
Perhaps at least use an `IllegalStateException()` here?
Alternatively you can use Kotlin's `error()` which does the same thing:
```suggestion
        val session = memory[sessionId] ?: error("Could not read faker memory for session id: $sessionId")
```